### DISCUSSION TITLE
generated pdf through PrintToPdfAsync, but the right and bottom of the layout are more blank

### DISCUSSION THREAD

**Discussion by shidaiman** [cefsharp/CefSharp#4380] [General]
I downloaded cefsharp.Offscreen (version 109.1.110)  and used code ‘PrintToPdfAsync’ to generate pdf from html pages, one elements in the html page is beyond the right side of the DIV element, I hid it through the style ‘overflow:hidden’.  When I view the generated pdf file，The whole content is automatically reduced，and there are some more blanks on the right and bottom of the content. The size of these blanks is just occupied by the hidden dom. I just want to get the content in DIV，how can I solve this problem? Thanks

**Comment by shidaiman** [comment]
If use lower version(103、104、105、106、107 or others）, it can display correctly, But once the html element goes beyond the bottom of the page, the element will be lost in pdf file!
（There are also bugs in version 108 or earlier）

Chrome web page effect:
![image](https://user-images.githubusercontent.com/10397126/215249950-25edca69-9ef7-49bb-ba2c-8639d1f88a4d.png)

Cefsharp  Output pdf file:
![image](https://user-images.githubusercontent.com/10397126/215250039-65ab894a-ee56-460e-8948-e7d5b7bc8a1b.png)

**Reply by amaitland** [reply]
Have you tried loading your `HTML` in `Chrome` and printing to `PDF` with similar settings?

`M108` brought upstream changes to the `PrintToPdf` implementation, it now uses the same underlying `Chrome DevTools Protocol` implementation. See https://github.com/chromiumembedded/cef/commit/25c75c5fc4fc7d7fbdebdf20ca56ab3162eab570

**Reply by shidaiman** [reply]
> Have you tried loading your `HTML` in `Chrome` and printing to `PDF` with similar settings?
> 
> `M108` brought upstream changes to the `PrintToPdf` implementation, it now uses the same underlying `Chrome DevTools Protocol` implementation. See [chromiumembedded/cef@25c75c5](https://github.com/chromiumembedded/cef/commit/25c75c5fc4fc7d7fbdebdf20ca56ab3162eab570)

thanks, I tried and the same bug occurred. It seems that it cannot be used for development

**Reply by amaitland** [reply]
Sounds like the problem needs to be fixed in `Chromium`. 

https://crbug.com/new

**Comment by shidaiman** [comment]
Dear gods, I'd appreciate some help...I have collapsed
